### PR TITLE
updates for openssl-1.1.1d

### PR DESCRIPTION
**Pull request by jmartin-r7**
Update openssl to version `1.1.1d`.

If `zlib` build integration is still desired add comments.

### CODE REVIEW

File: config/software/openssl.rb [hunk 1]
```diff
@@ -69,22 +77,18 @@
...
   configure_args = [
     "--prefix=#{install_dir}/embedded",
-    "--with-zlib-lib=#{install_dir}/embedded/lib",
-    "--with-zlib-include=#{install_dir}/embedded/include",
+    "no-comp",
     "no-idea",
     "no-mdc2",
     "no-rc5",
+    "no-ssl2",
+    "no-ssl3",
```
Comment (busterb) [inline config/software/openssl.rb]: Do we have to kill ssl3? That will mean Metasploit will no longer be able to connect to an SSLv3-only host, which, for instance, RDP can be.

File: config/software/openssl.rb [hunk 2]
```diff
@@ -69,22 +77,18 @@
...
+    "no-zlib",
```
Comment (busterb) [inline config/software/openssl.rb]: No argument with killing zlib, that's been not needed for a decade or so.

File: config/software/openssl.rb [hunk 3]
```diff
@@ -94,11 +98,6 @@
       "/bin/bash ./Configure solaris64-x86_64-gcc -static-libgcc"
     elsif omnios?
       "/bin/bash ./Configure solaris-x86-gcc"
```
Comment (busterb) [inline config/software/openssl.rb]: Note, a lot of this file came originally from Chef's upstreamin omnibus-software repository. If we delete stuff like this, it's usually nicer to sync with upstream.

Comment (busterb) [reply config/software/openssl.rb]: Ah, I see this is a sync with upstream!

File: config/software/openssl.rb (hunk 1 above)
Comment (jmartin-r7) [reply config/software/openssl.rb]: You are right, this came from upstream, I think we actually need to switch out for `enable-ssl3`.  Updating...

**Pull request merged**